app/models/book.rb

```diff
@@ -12,11 +12,20 @@ class Book < ApplicationRecord
   before_destroy :cascade
 
   validates :title, presence: true
-  validates :price, presence: true
   validates :description, presence: true
   validates :image_link, presence: true ,:format => URI::regexp(%w(http https))
   validates :isbn, uniqueness:  true
+  validates_numericality_of :isbn
```

Comment:
I thought before that the isbn is saved as string.
Anyway, the check length validator uses size method. I suppose it returns the number of bytes in memory not the number of digits. is that correct?

Reply:
it is a string of digits this validation check that [https://api.rubyonrails.org/classes/ActiveModel/Validations/HelperMethods.html#method-i-validates_numericality_of](url)

Reply:
Fair point. My bad, sorry